pkg/apis/core/validation/validation_test.go

```diff
@@ -780,6 +925,26 @@ func TestValidationOptionsForPersistentVolume(t *testing.T) {
 	}
 }
 
+func getCSIVolumeWithSecret(pv *core.PersistentVolume, secret *core.SecretReference, secretfield string) *core.PersistentVolume {
+	pvCopy := pv.DeepCopy()
+	if secret == nil {
+		return nil
+	}
```

Comment:
drop this, returning nil will panic the caller... passing a nil secret ref in should assign the nil secret ref as requested

Reply:
dropped this return. thanks @liggitt